### DISCUSSION TITLE
Refresh content after automation on-click action

### DISCUSSION THREAD

**Discussion by diogenesbrussels** [Budibase/budibase#3985] [Help]
Hello,

I'm trying to refresh the content of the current page after an automation is completed.
More details on my usage:
- I would like to create a row when the user clicks on a button
- The automation works well and create the row with the right pre populated fields
- After the automation is completed the new content is displayed on the page

I was unable to refresh the page or the content with the new content.
I tried the following actions:
- Refresh Data Provider (doesn't wait for automation?)
- Navigate To : same page url (doesn't do anything?)
- Navigate To : same page url + #currentsection (doesn't do anything?)
- Navigate To : same page url + ?updatedsomething=1 (doesn't do anything?)

Is there another way to do this or the force the refresh?
Suggestions/ideas: add an async option in the automation action ? A force refresh in the Navigate To action?
I may be missing something or another way to do it, let me know.

Thank you!

**Comment by mjashanks** [comment]
Automation work completely asynchronously. They return to the client immediately whether they have completed or not.

So, this is not possible at the moment. 

I believe that there would be quite a lot of work involved in allowing an automation to run synchronously - though I agree that this would be a nice feature as well, and up-level the power of automations.

Currently, we have an action called "Refresh Data Provider" (below). I think we could add a "delay" to this action. I.e. "Refresh Data provider after 5 seconds".

**Reply by diogenesbrussels** [reply]
@mjashanks Thanks for the information, synchronous automations would really be a great option but I understand that it takes a lot of work.
Adding a delay for the refresh could be a solution but maybe adding a 'launch js script' action could be more flexible and powerful to do the same thing?

**Reply by mjashanks** [reply]
What would the JS script do in this instance? Do you mean something like...

```javascript
setTimeout(function() {
  actions.RefreshDataProvider("My Data Provider");
}, 5000);
```

**Reply by diogenesbrussels** [reply]
Yes, that would be great. I didn't know I could access the actions in this way, is there any documentation on this?
I would have just done a location.reload() or similar.

**Reply by mjashanks** [reply]
No - sorry, you can't! I just made that up - I think it would be a good idea to implement this.

**Reply by diogenesbrussels** [reply]
😀 Yes, it would be a good idea!
Wouldn't it be even better if:
`actions.TriggerAutomation("au_66d8d90c89e3493b8ec6719fd43f3286", function () {
  actions.RefreshDataProvider("My Data Provider");
});`
What do you think?

**Reply by mjashanks** [reply]
totally agree, but with `async`/`await`!

Just opening this flow up to Javascript would close any gaps in this.

**Reply by davebe1** [reply]
Having the same issue. I need a component on the DOM to update with the information once an async automation completes.
